src/docfx/build/metadata/ContributionProvider.cs

```diff
@@ -68,6 +68,15 @@ internal class ContributionProvider
             foreach (var commit in contributionCommits)
             {
                 var (error, githubUser) = _githubAccessor.GetUserByEmail(commit.AuthorEmail, repoOwner, repoName, commit.Sha);
+                var username = GetGitHubUsernameWhenCommitWithNoReplyEmail(commit.AuthorEmail);
```

Comment:
move into `GetUserByEmail`?

Reply:
If this email pattern matches, we could save the API call to GitHub.